### DISCUSSION TITLE
How to control which toolchain is used?

### DISCUSSION THREAD

**Discussion by rsgrimes** [hathach/tinyusb#1563] [Q&A] [answered]
Trying to build examples on Linux, but the wrong toolchain is being chosen; I have several versions of GCC on my machine, and I need it to use the newer version (the old version is needed, and I can't remove it). Where does the build system pull in this information?

**Comment by hathach** [comment, marked as answer]
the answer is always within the source. arm-none-eabi-gcc are used by default, try to determine it with 

```shell
which arm-none-eabi-c
```

You could move the one you want to use up in the PATH environment order, or just pass its path via `CROSS_COMPILE=` in the make command 
 https://github.com/hathach/tinyusb/blob/2110e276783e2f97f2a942b082355f1d4a181573/examples/make.mk#L51-L60

**Reply by rsgrimes** [reply]
CROSS_COMPILE works fine.

**Comment by rsgrimes** [comment]
Understood, thanks, the CROSS_COMPILE= switch passed to the make command in the example project directory seems to pull in the desired toolchain, thanks!
